### DISCUSSION TITLE
Hash Signing

### DISCUSSION THREAD

**Discussion by heruwaspodov** [gettalong/hexapdf#330] [Q&A] [answered]
Hi @gettalong 
I will implement the hash signing. Is it correct to use the `external_signing` to execute?

**Comment by gettalong** [comment, marked as answer]
Yes, see https://hexapdf.gettalong.org/documentation/digital-signatures/signing-pdfs-howto.html#external for details.

**Comment by heruwaspodov** [comment]
Hi mas @gettalong 

i tried to sign with hash signing
this code is like the gist below:
https://gist.github.com/heruwaspodov/869ba59632178621e22cca1ffd05cbbe

to exec

ann = Annotations::AnnotationSignatureHash.new('Variable.pdf')
ann.call

the result below

<img width="1680" alt="image" src="https://github.com/user-attachments/assets/de7da4db-717b-4c4d-8049-2b76183b7f53">

the certificate is not related/references to the signature field

could you give me advice, mas?
thank you

**Reply by gettalong** [reply]
I have put my comments on the referenced gist.
